Toggl.Ultrawave.Tests.Integration/WorkspacesClientTests.cs

```
@@ -0,0 +1,75 @@
+﻿using System;
+using System.Linq;
+using System.Reactive.Linq;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Toggl.Ultrawave.Clients;
+using Toggl.Ultrawave.Exceptions;
+using Xunit;
+
+using WorkspaceHelper = Toggl.Ultrawave.Tests.Integration.Helper.Workspace;
+
+namespace Toggl.Ultrawave.Tests.Integration
+{
+    public class WorkspacesClientTests
+    {
+        public class TheGetMethod
+        {
+            private readonly IWorkspacesClient workspacesClient = new TogglClient(ApiEnvironment.Staging).Workspaces;
+
+            [Fact]
+            public async Task WorksForAllWorkspaces()
```

This test is very broken. The action in fact throws an exception, but because the test only checks for ApiException, this is not caught, and the test 'succeeds'.

We can remove the generic constraint from the ShouldNotThrow to catch all exceptions (and we should in general do so in probably all cases).

However, this still ends up with nested exceptions which is bad output for the test. Not to mention that this is now following AAA. All assertions should be at the end, not inside the acting step.

@willsb feel free to comment.